source/common/upstream/load_balancer_config.h

```diff
@@ -0,0 +1,281 @@
+#pragma once
+
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/extensions/load_balancing_policies/common/v3/common.pb.h"
+#include "envoy/extensions/load_balancing_policies/least_request/v3/least_request.pb.h"
+#include "envoy/extensions/load_balancing_policies/least_request/v3/least_request.pb.validate.h"
+#include "envoy/extensions/load_balancing_policies/maglev/v3/maglev.pb.h"
+#include "envoy/extensions/load_balancing_policies/maglev/v3/maglev.pb.validate.h"
+#include "envoy/extensions/load_balancing_policies/random/v3/random.pb.h"
+#include "envoy/extensions/load_balancing_policies/random/v3/random.pb.validate.h"
+#include "envoy/extensions/load_balancing_policies/ring_hash/v3/ring_hash.pb.h"
+#include "envoy/extensions/load_balancing_policies/ring_hash/v3/ring_hash.pb.validate.h"
+#include "envoy/extensions/load_balancing_policies/round_robin/v3/round_robin.pb.h"
+#include "envoy/extensions/load_balancing_policies/round_robin/v3/round_robin.pb.validate.h"
+#include "envoy/extensions/load_balancing_policies/subset/v3/subset.pb.h"
+#include "envoy/extensions/load_balancing_policies/subset/v3/subset.pb.validate.h"
+#include "envoy/upstream/load_balancer.h"
+
+namespace Envoy {
+namespace Upstream {
+
+/**
+ * Load balancer config that used to wrap the legacy proto config.
+ */
+class LegacyTypedRoundRobinLbConfig : public LoadBalancerConfig {
```

Comment:
so I'm (tentatively) ok with this for this PR, but I'm inclined to think that we should move all ring hash code (legacy and otherwise) to the ring hash extension, and same for all the others.
I think we'd want each factory having a "convert config" function and if we had legacy ring hash config, the function would create new style config, and for new style configs the convert function would be a no-op.  WdYT?

Reply:
I believe you're correct but can you explain why not?  Even if we can only move the others out it's still worth doing IMO
Sorry I do owe you a full pass but I'm on reduced hours this week - I'll try to do a full pass tomorrow and thanks for your patience.  Feel free to resassign if my turnaround is too slow :-(

Reply:
> I believe you're correct but can you explain why not? Even if we can only move the others out it's still worth doing IMO

The `LoadBalancerSubsetInfoImpl` is referenced by the core old directly for now in the legacy code path. So, we cannot move it to the `extensions/` for now.
But I do agree you point, so I have moved others. :)

Reply:
> Sorry I do owe you a full pass but I'm on reduced hours this week - I'll try to do a full pass tomorrow and thanks for your patience. Feel free to resassign if my turnaround is too slow :-(

Just take it easy. I am not in a hurry. And rather than to land this patch quickly, I prefer to make it better. I am willing to give it more time.